numpy/lib/tests/test_histograms.py

```diff
@@ -324,8 +324,10 @@ def test_some_nan_values(self):
 
     def test_datetime(self):
         begin = np.datetime64('2000-01-01', 'D')
-        offsets = np.array([0, 0, 1, 1, 2, 3, 5, 10, 20])
-        bins = np.array([0, 2, 7, 20])
+        # NOTE: this test is less useful with new prohibition on
+        # int64 -> m8 casting
+        offsets = np.array([0, 0, 1, 1, 2, 3, 5, 10, 20], dtype='m8[D]')
+        bins = np.array([0, 2, 7, 20], dtype='m8[D]')
         dates = begin + offsets
         date_bins = begin + bins
```

Comment:
This doesn't look right - I think you should leave what you touched unchanged, and it should read
```
dates = begin + offsets.astype('m8[D]')
date_bins = begin + bins.astype('m8[D]')
```